src/server/saved_objects/service/lib/schema.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export class SavedObjectsSchema {
+  _namespaceAgnosticTypes = [];
+
+  addNamespaceAgnosticType(type) {
```

Comment:
Can we avoid making the `Schema` mutable like this and instead require that namespace agnostic types are passed into the `Schema` constructor? If we make it mutable like this then I'm not really sure how we'll know when we can reliably check for namespace agnostic types, or other data that ends up in this class (like "mappings") as it could have types added at any time...

Reply:
We definitely can, I assume this would require modifications to the plugin specifications to allow plugins to declaratively add these?

Reply:
It looks like the mappings are part of the` uiExports` currently, would you prefer that the we add a `savedObjectsSchema` to the `uiExports` or would you prefer a separate entry added to the plugin specification?

Reply:
I'd love if we unified the mappings and metadata like this into a `savedObjectsSchema` uiExport type of some sort.